Doc/glossary.rst

```diff
@@ -942,8 +1023,11 @@ Glossary
       See :term:`method resolution order`.
 
    mutable
-      Mutable objects can change their value but keep their :func:`id`.  See
-      also :term:`immutable`.
+      Mutable objects can change their value but keep their :func:`id`.
+      In multi-threaded programs, mutable objects that are shared between
+      threads require careful synchronization to avoid :term:`concurrent modification`
+      issues.  See also :term:`immutable`, :term:`thread-safe`, and
```

Comment:
The first sentence is a run-on and should be split into two. However, I think the text you inherited here is confusing and should perhaps be reworded to something like ``An :term:`object` with state that is allowed to change during the course of a program.`` The id of the object isn't really what's important and distracts from the main point: changing internal state